pkg/kubelet/kubelet.go

```diff
@@ -1267,13 +1268,27 @@ func (kl *Kubelet) SyncPods(pods []api.BoundPod) error {
 		pc := podContainer{podFullName, uid, containerName}
 		if _, ok := desiredContainers[pc]; !ok {
 			glog.V(1).Infof("Killing unwanted container %+v", pc)
-			err = kl.killContainer(container)
+			err = kl.killContainer(dockerContainers[ix])
 			if err != nil {
 				glog.Errorf("Error killing container %+v: %v", pc, err)
+			} else {
+				killed = append(killed, dockerContainers[ix].ID)
 			}
 		}
 	}
 
+	for {
+		running, err := dockertools.GetRunningContainers(kl.dockerClient, killed)
+		if err != nil {
+			glog.Errorf("Failed to poll container state: %v", err)
+		}
+		if len(running) == 0 {
+			break
+		}
+		glog.Infof("Some containers still running: %v", running)
+		time.Sleep(time.Second)
+	}
```

Comment:
or fire them off in goroutines?  Isn't this what go is supposed to be good at?  what will docker do itf we async blast a bunch of kills at it?